### PR TITLE
Change suffix of upgrade steps functions from plan to step

### PR DESCRIPTION
**Pull request by gabrielcocenza**
- removed unnecessary parallel parameter.

### CODE REVIEW

File: cou/apps/auxiliary.py [hunk 1]
```diff
@@ -139,31 +139,28 @@ class CephMonApplication(OpenStackAuxiliaryApplication):
...
+        :return:  List of pre upgrade steps.
         :rtype: list[PreUpgradeStep]
         """
-        return super().pre_upgrade_plan(target) + [self._get_change_require_osd_release_plan()]
+        return super().pre_upgrade_steps(target) + [self._get_change_require_osd_release_step()]
 
-    def _get_change_require_osd_release_plan(self, parallel: bool = False) -> PreUpgradeStep:
-        """Get plan to set correct value for require-osd-release option on ceph-mon.
+    def _get_change_require_osd_release_step(self) -> PreUpgradeStep:
+        """Get the ste to set correct value for require-osd-release option on ceph-mon.
```
Comment (agileshaw) [inline cou/apps/auxiliary.py:153]:
```suggestion
        """Get the step to set correct value for require-osd-release option on ceph-mon.
```

File: cou/apps/base.py [hunk 2]
```diff
@@ -496,42 +496,35 @@ def generate_upgrade_plan(self, target: OpenStackRelease) -> ApplicationUpgradeP
...
+            + self.post_upgrade_steps(target)
         )
         for step in all_steps:
             if step:
                 upgrade_steps.add_step(step)
         return upgrade_steps
 
-    def _get_upgrade_current_release_packages_plan(self, parallel: bool = False) -> PreUpgradeStep:
+    def _get_upgrade_current_release_packages_step(self) -> PreUpgradeStep:
         """Get Plan for upgrading software packages to the latest of the current release.
```
Comment (agileshaw) [inline cou/apps/base.py:509]:
```suggestion
        """Get step for upgrading software packages to the latest of the current release.
```

File: cou/apps/base.py [hunk 3]
```diff
@@ -496,42 +496,35 @@ def generate_upgrade_plan(self, target: OpenStackRelease) -> ApplicationUpgradeP
...
             ),
-            parallel=parallel,
             coro=upgrade_packages(self.status.units.keys(), self.model, self.packages_to_hold),
         )
 
-    def _get_refresh_charm_plan(
-        self, target: OpenStackRelease, parallel: bool = False
-    ) -> PreUpgradeStep:
+    def _get_refresh_charm_step(self, target: OpenStackRelease) -> PreUpgradeStep:
         """Get plan for refreshing the current channel.
```
Comment (agileshaw) [inline cou/apps/base.py:522]:
```suggestion
        """Get step for refreshing the current channel.
```

File: cou/apps/base.py [hunk 4]
```diff
@@ -433,26 +433,26 @@ async def _check_upgrade(self, target: OpenStackRelease) -> None:
...
         return [
-            self._get_upgrade_current_release_packages_plan(),
-            self._get_refresh_charm_plan(target),
+            self._get_upgrade_current_release_packages_step(),
+            self._get_refresh_charm_step(target),
         ]
 
-    def upgrade_plan(self, target: OpenStackRelease) -> list[UpgradeStep]:
-        """Upgrade planning.
+    def upgrade_plan_steps(self, target: OpenStackRelease) -> list[UpgradeStep]:
```
Comment (agileshaw) [inline cou/apps/base.py:449]:
```suggestion
    def upgrade_steps(self, target: OpenStackRelease) -> list[UpgradeStep]:
```

File: cou/apps/base.py (hunk 3 above)
Comment (agileshaw) [reply cou/apps/base.py:522]: And there are other similar functions require this change

File: cou/apps/channel_based.py [hunk 5]
```diff
@@ -60,16 +60,16 @@ def is_versionless(self) -> bool:
         """
         return not all(unit.workload_version for unit in self.status.units.values())
 
-    def post_upgrade_plan(self, target: OpenStackRelease) -> list[PostUpgradeStep]:
+    def post_upgrade_steps(self, target: OpenStackRelease) -> list[PostUpgradeStep]:
         """Post Upgrade planning.
```
Comment (agileshaw) [inline cou/apps/channel_based.py:64]:
```suggestion
        """Post Upgrade steps planning.
```

File: cou/apps/subordinate.py [hunk 6]
```diff
@@ -25,33 +25,33 @@
...
-        :return: Plan that will add pre upgrade as sub steps.
+        :return: List of pre upgrade steps.
         :rtype: list[PreUpgradeStep]
         """
-        return [self._get_refresh_charm_plan(target)]
+        return [self._get_refresh_charm_step(target)]
 
-    def upgrade_plan(self, target: OpenStackRelease) -> list[UpgradeStep]:
-        """Upgrade planning.
+    def upgrade_plan_steps(self, target: OpenStackRelease) -> list[UpgradeStep]:
```
Comment (agileshaw) [inline cou/apps/subordinate.py:38]:
```suggestion
    def upgrade_steps(self, target: OpenStackRelease) -> list[UpgradeStep]:
```

**Review by agileshaw — CHANGES_REQUESTED** [review]

**Review by rgildein — CHANGES_REQUESTED** [review]
This should against main branch and not dev branch, we will merge main to dev, but we want to minimize lines for review for merger of dev/data-plane to main.

**Comment by gabrielcocenza** [comment]
It was asked to land those changes on main. The #235 was opened to this.

**Pull request closed without merging**